ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,664 @@
+import itertools
+import warnings
+from collections import defaultdict
+from typing import Any, Callable, cast, Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+from typing_extensions import Literal
+
+import ignite.distributed as idist
+from ignite.distributed.utils import _all_gather_tensors_with_shapes
+from ignite.metrics.metric import reinit__is_reduced
+from ignite.metrics.recall import _BasePrecisionRecall
+from ignite.utils import to_onehot
+
+
+class MeanAveragePrecision(_BasePrecisionRecall):
+    _tp: Dict[int, List[torch.Tensor]]
+    _fp: Dict[int, List[torch.Tensor]]
+    _scores: Union[Dict[int, List[torch.Tensor]], List[torch.Tensor]]
+    _P: Union[Dict[int, int], List[torch.Tensor]]
+
+    def __init__(
+        self,
+        rec_thresholds: Optional[Union[Sequence[float], torch.Tensor]] = None,
+        average_operand: Optional[Literal["precision", "max-precision"]] = "precision",
+        class_mean: Optional[Literal["micro", "macro", "weighted", "with_other_dims"]] = "macro",
+        classification_is_multilabel: bool = False,
+        allow_multiple_recalls_at_single_threshold: bool = False,
```

Comment:
```suggestion
        allow_multiple_recalls: bool = False,
```
Same here, the name is too long.

Reply:
I introduced this arg only for the sake of pycocotools.

During computing recalls and precisions, one threshold(one score) yields exactly one recall and precision so we would have #unique_scores thresholds, but pycocotools has not put effort to select unique scores and one score yields as many as the predictions that have that score. This is wrong but since we wanted to adhere to pycocotools, I added this arg. Now I think it’s better to override `_measure_recall_and_precision` and `_measure_average_precision` in `ObjectDetectionMap` and remove this arg. Am I right?